src/type_level_bool.rs

```diff
@@ -0,0 +1,30 @@
+#[derive(Copy, Clone, Debug)]
+pub struct True;
+#[derive(Copy, Clone, Debug)]
+pub struct False;
+
+pub trait ToBool {
+    type Value;
+    fn to_bool(_: Self) -> Self::Value;
+}
+impl ToBool for [(); 0] {
+    type Value = False;
+    fn to_bool(_: Self) -> Self::Value {
+        False
+    }
+}
+impl ToBool for [(); 1] {
+    type Value = True;
+    fn to_bool(_: Self) -> Self::Value {
+        True
+    }
+}
+
+/// Converts a `const bool` to a type-level boolean.
+#[macro_export]
+macro_rules! to_bool {
+    ($x:expr $(,)?) => {{
+        const ASSERT: bool = $x;
+        $crate::type_level_bool::ToBool::to_bool([(); ASSERT as usize])
+    }};
+}
```

Comment:
I don't want to export this macro publicly. Please make this `#[doc(hidden)]` and prefix it with an underscore (or two).

All macros that use this should have `#[macro_export(local_inner_macros)]`.

Also, the `$(,)?` seems unnecessary.

Reply:
The module itself is `doc_hidden`, so I assume this hides everything in the module ?

Reply:
Ah no, that's only with rust2018 probably

Reply:
Macros and docs are weird. For example, `#![deny(missing_docs)]` doesn't work with them. Not sure if 2018 edition changes anything there.